common/components/Header/components/CustomNodeModal.tsx

```diff
@@ -0,0 +1,260 @@
+import React from 'react';
+import classnames from 'classnames';
+import Modal, { IButton } from 'components/ui/Modal';
+import translate from 'translations';
+import { NETWORKS, CustomNodeConfig } from 'config/data';
+
+const NETWORK_KEYS = Object.keys(NETWORKS);
+
+interface Props {
+  handleAddCustomNode(node: CustomNodeConfig): void;
+  handleClose(): void;
+}
+
+interface State {
+  name: string;
+  url: string;
+  port: string;
+  network: string;
+  hasAuth: boolean;
+  username: string;
+  password: string;
+  customEip155: boolean;
+  customChainId: string;
+}
+
+interface Input {
+  name: string;
+  label: any;
+  placeholder: string;
+}
+
+export default class CustomNodeModal extends React.Component<Props, State> {
+  public state: State = {
+    name: '',
+    url: '',
+    port: '',
+    network: NETWORK_KEYS[0],
+    customEip155: false,
+    customChainId: '',
+    hasAuth: false,
+    username: '',
+    password: '',
+  };
+
+  public render() {
+    const { handleClose } = this.props;
+    const isHttps = window.location.protocol.includes('https');
+    const invalids = this.getInvalids();
+
+    const createInput = (name: string, placeholder?: string, type?: string) => {
```

Comment:
Could this take an object of `name` `placeholder?` and `type?` instead? Then you can just spread the object inside `input`

Reply:
Also you can use an implict return here

Reply:
And it might be worth extracting this function into a method so its not being reinstantiated on each re-render